tests/ut/RialtoGstTest.cpp

```diff
@@ -0,0 +1,174 @@
+    std::call_once(onceFlag,
+                   [this]()
+                   {
+                       expectSinksInitialisation();
+                       gst_init(nullptr, nullptr);
+                       const auto registerResult =
+                           gst_plugin_register_static(GST_VERSION_MAJOR, GST_VERSION_MINOR, "rialtosinks",
+                                                      "Sinks which communicate with RialtoServer", rialto_mse_sinks_init,
+                                                      "1.0", "LGPL", PACKAGE, PACKAGE, "http://gstreamer.net/");
+                       EXPECT_TRUE(registerResult);
+                   });
+}
+
+RialtoGstTest::~RialtoGstTest()
+{
+    testing::Mock::VerifyAndClearExpectations(&m_controlFactoryMock);
+}
+
+std::size_t RialtoGstTest::ReceivedMessages::size() const
+{
+    return m_receivedMessages.size();
+}
+
+bool RialtoGstTest::ReceivedMessages::empty() const
+{
+    return m_receivedMessages.empty();
+}
+
+bool RialtoGstTest::ReceivedMessages::contains(const GstMessageType &type) const
+{
+    return std::find(m_receivedMessages.begin(), m_receivedMessages.end(), type) != m_receivedMessages.end();
+}
+
+RialtoMSEBaseSink *RialtoGstTest::createAudioSink() const
+{
+    EXPECT_CALL(*m_controlFactoryMock, createControl()).WillOnce(Return(m_controlMock));
+    EXPECT_CALL(*m_controlMock, registerClient(_, _))
+        .WillOnce(DoAll(SetArgReferee<1>(ApplicationState::RUNNING), Return(true)));
+    GstElement *audioSink = gst_element_factory_make("rialtomseaudiosink", "rialtomseaudiosink");
+    return RIALTO_MSE_BASE_SINK(audioSink);
+}
+
+RialtoMSEBaseSink *RialtoGstTest::createVideoSink() const
+{
+    EXPECT_CALL(*m_controlFactoryMock, createControl()).WillOnce(Return(m_controlMock));
+    EXPECT_CALL(*m_controlMock, registerClient(_, _))
+        .WillOnce(DoAll(SetArgReferee<1>(ApplicationState::RUNNING), Return(true)));
+    GstElement *videoSink = gst_element_factory_make("rialtomsevideosink", "rialtomsevideosink");
+    return RIALTO_MSE_BASE_SINK(videoSink);
+}
+
+GstElement *RialtoGstTest::createPipelineWithSink(RialtoMSEBaseSink *sink) const
+{
+    GstElement *pipeline = gst_pipeline_new("test-pipeline");
+    gst_bin_add(GST_BIN(pipeline), GST_ELEMENT_CAST(sink));
+    return pipeline;
+}
+
+RialtoGstTest::ReceivedMessages RialtoGstTest::getMessages(GstElement *pipeline) const
+{
+    RialtoGstTest::ReceivedMessages result;
+    GstBus *bus = gst_element_get_bus(pipeline);
+    if (!bus)
+    {
+        return result;
+    }
+    GstMessage *msg{gst_bus_pop(bus)};
+    while (msg)
+    {
+        result.m_receivedMessages.push_back(GST_MESSAGE_TYPE(msg));
+        gst_message_unref(msg);
+        msg = gst_bus_pop(bus);
+    }
+    gst_object_unref(bus);
+    return result;
+}
+
+void RialtoGstTest::setPausedState(GstElement *pipeline, RialtoMSEBaseSink *sink)
+{
+    EXPECT_CALL(m_mediaPipelineMock, attachSource(_)).WillOnce(Return(true));
```

Comment:
Do you think its worth/possible to test some of the input parameters here in load(_, _, _) and createMediaPipeline(_, _)?

Reply:
MediaPipeline calls with params are tested in MediaPlayerClientBackendTests, but I can add param checking here as well. If it's not a problem, I will do it in new pull request, as I'm still working on adding new rialto-gstreamer unit tests.